pkg/raft/log.go

```diff
@@ -54,6 +60,10 @@ type raftLog struct {
 	// they will be saved into storage.
 	unstable unstable
 
+	// termCache contains a suffix of the whole raftLog(both stable and unstable)
+	// used for term lookup.
+	termCache *TermCache
```

Comment:
Since it's always populated and non-empty, let's not make it a pointer / allocation.

nit: we don't store necessarily a whole raft log suffix in the cache

```suggestion
	// termCache contains a compressed information about entry terms for a
	// suffix of the raftLog. Covers entries both in unstable and storage.
	termCache TermCache
```